oneview_redfish_toolkit/connection.py

```diff
@@ -31,76 +32,63 @@
-            ServiceRoot blueprint. If true, even if authentication_mode is
-            set to session it will use the information on the conf file to
-            return a connection.  This is a workaround to allow ServiceRoot
-            to retrieve the appliance UUID before user logs in.
+    auth_mode = config.get_authentication_mode()
+    ov_config = None
 
-        Returns:
-            OneViewClient object
+    if auth_mode == "conf":
+        return get_oneview_client_conf_mode(ip_oneview, api_version)
 
-        Exceptions:
-            HPOneViewException if can't connect or reconnect to OV
-    """
+    if auth_mode == "session":
+        ov_config = create_oneview_config(ip=ip_oneview,
+                                          api_version=api_version,
+                                          token=token)
 
-    auth_mode = config.get_authentication_mode()
+    try:
+        oneview_client = OneViewClient(ov_config)
+        return oneview_client
+    except Exception:
+        logging.exception("Failed to recover session based connection")
+        raise
 
-    if auth_mode == "conf" or is_service_root:
-        # Doing conf based authentication
-        ov_config = create_oneview_config(ip=config.get_oneview_ip(),
-                                          credentials=config.get_credentials())
-        ov_client = None
 
-        # Check if connection is ok yet
-        try:
-            # Check if OneViewClient already exists
-            if 'ov_client' not in globals():
-                globals()['ov_client'] = OneViewClient(ov_config)
-
-            ov_client = globals()['ov_client']
-            ov_client.connection.get('/rest/logindomains')
-            return ov_client
-        # If expired try to make a new connection
-        except Exception:
-            try:
-                logging.exception('Re-authenticated')
-                if ov_client:
-                    ov_client.connection.login(config.get_credentials())
-                    return ov_client
-            # if failed abort
-            except Exception:
-                raise
-    else:
-        # Auth mode is session
-        ov_config = create_oneview_config(ip=config.get_oneview_ip(),
-                                          session_token=session_id)
+def get_oneview_client_conf_mode(ip_oneview, api_version=None):
+    ov_client = None
+    # Check if connection is ok yet
+    try:
+        ov_config = \
+            create_oneview_config(ip=ip_oneview,
+                                  api_version=api_version,
+                                  credentials=config.get_credentials())
+
+        ov_client = OneViewClient(ov_config)
+        return ov_client
+    except Exception:
         try:
-            oneview_client = OneViewClient(ov_config)
-            oneview_client.connection.get('/rest/logindomains')
-            return oneview_client
+            logging.exception('Re-authenticated')
+            if ov_client:
```

Comment:
Not sure if we need to re-authenticate since we just created the cliente

Reply:
Yes, we do not need that anymore, once we removed the cached connection for "conf" mode we can remove this.